File: C#JAVA_Client_Server/ClientServcerC#Protocols/Solution1/Domain/Validators/CursaValidator.cs
using System;
using Domain.validators.interfaceAndException;


namespace Domain.validators
{
    public class CursaValidator : IValidator<Cursa>
    {
        public void validate(Cursa entity)
        {
            if (entity.NumeCupa.Length == 0 || entity.CapacitateMotor.Equals(null))
            {
                throw new Exception("Cursa INVALIDA!");
            }
        }
    }
}

File: C#JAVA_Client_Server/ClientServcerC#Protocols/LongExample/Domain/src/main/java/entities/domain/validator/CursaValidator.java
package entities.domain.validator;


import entities.domain.Cursa;
import entities.domain.validator.interfaceAndExceptions.ValidationException;
import entities.domain.validator.interfaceAndExceptions.Validator;

public class CursaValidator implements Validator<Cursa> {
    @Override
    public void validate(Cursa entity) throws ValidationException {
        if(entity.getCapacitateMotor()==null||entity.getNumeCupa().length()==0){
            throw new ValidationException("entities.domain.Cursa invalida");
        }
    }
}
